### ISSUE TITLE
enhancement - ifsc service

### ISSUE THREAD

**Issue by SaiSkandaTNI** [knadh/dns.toys#81]
I made this ifsc service as a POC on my fork - https://github.com/SaiSkandaTNI/dns.toys

Since I am not a developer, I did not initate a pull request (I used Claude to make the changes in the fork above)

And if this were to be actually implemented on dns.toys, caching should be implemented which I did not in my fork... The source for this is Razorpay who has open sourced this and the data can be fetched and cached from their GitHub repo.

If this is something poeple might find useful and someone can clean up my code and implement caching, that would be great!

**Example** - dig +short KKBK0008122.ifsc @127.0.0.1 -p 5354 TXT

**Reponse**

"BANK: Kotak Mahindra Bank"
"BRANCH: BELLANDUR BRANCH"
"CITY: BANGALORE"
"STATE: KARNATAKA"

**Why build this?**

We have users who work in remote areas with little mobile coverage, our app is designed to work offline but IFSC validation needs to be up to date so thought performing this via DNS might make this work in the worst of cellular reception areas. At least thats my guess.

side note - if someone has any suggestions on a better way to validate IFSC / Account number - would love to hear it!

**Comment by knadh** [comment]
Adding IFSC lookup is a good idea. It's not ideal to hit an upstream API for every request though. Like some of the other services (dict), the data dump can be read from the disk locally and loaded into the memory to respond to requests. The dump can be updated periodically externally via a chron or something.

**Comment by SaiSkandaTNI** [comment]
Yes Agreed, I posted this here in the hope that someone with the technical know how to try and set up the dump and build the logic to extract the relevent data from the RazorPay dump.

Both the Upstream API code and Raw data are available on the RazorPay GitHub and are actively updated - https://ifsc.razorpay.com/

Thanks for creating this @knadh - really good to see people still think about low badwidth, when everyone is going towards 5G and beyond.

A lot of people in poverty live in low bandwith areas and we need more open source tools like this to help them.
